File: Scala/src/main/scala/Histogram.scala
// https://www.codewars.com/kata/59f44c7bd4b36946fd000052/scala

object Histogram
{
	def hist(s: String): String =
	{
		var strings: Array[String] = Array()
		for (letter <- "uwxz")
		{
			val count = s.count(c => c.equals(letter))
			if (count > 0)
			{
				val string = "%c  %-5d %s".format(letter, count, "*" * count)
				strings :+= string
			}
		}
		strings.mkString("\n")
	}
}

File: Ruby/src/histogram.rb
# https://www.codewars.com/kata/59f44c7bd4b36946fd000052/ruby

class Histogram
	public
	
	def self.hist(s)
		strings = []
		%w[u w x z].each do |letter|
			count = s.scan(letter).length
			if count < 1
				next
			end
			string = "%c  %-5d %s" % [letter, count, "*" * count]
			strings.push string
		end
		strings * "\n"
	end
end
